### ISSUE TITLE
Interpolate with tuning getting killed

### ISSUE THREAD

**Issue by joemathai** [kpu/kenlm#130]
I am able to interpolate when the datasets used are smaller in size but I get the following error otherwise. I am fairly certain that it is not an issue with disk space. 
```
Processing model 0/4: data1
----5---10---15---20---25---30---35---40---45---50---55---60---65---70---75---80---85---90---95--100
####################################################################################################
Processing model 1/4: data2
----5---10---15---20---25---30---35---40---45---50---55---60---65---70---75---80---85---90---95--100
##########################################################################################+++++++++Killed
```

**Comment by kpu** [comment]
That's the OOM killer.  I was bad about optimizing the interpolate program's memory consumption (and in particular -S is a lie).  Try adding -S 1G.

**Comment by joemathai** [comment]
I am getting the following output when trying to interpolate with lower memory.
```
Iteration 0: weights = 0.25 0.25 0.25 0.25
----5---10---15---20---25---30---35---40---45---50---55---60---65---70---75---80---85---90---95--100
0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
Perplexity = 40.863
Iteration 1: weights = -0.531328 4.96237 -4.80552 1.387
----5---10---15---20---25---30---35---40---45---50---55---60---65---70---75---80---85---90---95--100
****************************************************************************************************
Perplexity = inf
Iteration 2: weights = -nan -nan -nan -nan
----5---10---15---20---25---30---35---40---45---50---55---60---65---70---75---80---85---90---95--100
****************************************************************************************************
Perplexity = -nan
Iteration 3: weights = -nan -nan -nan -nan
----5---10---15---20---25---30---35---40---45---50---55---60---65---70---75---80---85---90---95--100
***************
```

**Comment by kpu** [comment]
Hmmm.  Try changing the learning rate 0.7 to a smaller number on this line:

https://github.com/kpu/kenlm/blob/4d07012267234276384bd7e6809cc131910943be/lm/interpolate/tune_weights.cc#L29

Then recompile.  As you can probably infer, interpolation was written around the time neural networks took over, so it's received less tender love and care.  Sorry.

**Comment by joemathai** [comment]
@kpu thanks!
